source/game.database.bmx

```diff
@@ -235,12 +232,96 @@ Type TDatabaseLoader
 			Notify "Important data is missing:  series:"+totalSeriesCount+"  movies:"+totalMoviesCount+"  news:"+totalNewsCount+"  adcontracts:"+totalContractsCount
 		EndIf
 
+		LoadDatabaseLocalizations(dbDirectory)
 
 		'fix potentially corrupt data
 		FixLoadedData()
 	End Method
-	
-	
+
+
+	Function LoadDatabaseLocalizations(dbDirectory:String)
+		Local langDir:String = dbDirectory+"/lang/"
+		Local dbl:TDatabaseLocalizer = GetDatabaseLocalizer()
+		Local toStore:TPersonLocalization[] = new TPersonLocalization[100]
+
+		For Local l:TLocalizationLanguage = EachIn TLocalization.languages
+			Local code:String = l.languageCode
+			Local file:String = langDir + code+".xml"
+			If FileType(file) = 1
+				Local xml:TXmlHelper = TXmlHelper.Create(file)
+				Local allGlobalVars:TxmlNode = xml.FindRootChildLC("globalvariables")
+				If allGlobalVars
+					Local gvl:TLocalizationLanguage = dbl.getGlobalVariables(code)
+					If Not gvl
+						gvl = new TLocalizationLanguage
+						gvl.languageCode = code
+						dbl.globalVariables.insert(code, gvl)
+					EndIf
+					Local varName:String
+					Local value:String
+					For Local varNode:TxmlNode = EachIn xml.GetNodeChildElements(allGlobalVars)
+						varName = varNode.getName().toLower()
+						value = varNode.getContent()
+						If varName And value Then gvl.map.insert(varName, value)
+					Next
+				EndIf
+
+				Local nodeAllPersons:TxmlNode
+				nodeAllPersons = xml.FindRootChildLC("persons")
+				Local index:Int = 0
+				Local personCollection:TPersonBaseCollection = GetPersonBaseCollection()
+				For Local nodePerson:TxmlNode = EachIn xml.GetNodeChildElements(nodeAllPersons)
+					If nodePerson.getName() <> "person" Then Continue
+					Local data:TData = New TData
+					xml.LoadValuesToData(nodePerson, data, ["guid","first_name", "last_name", "nick_name", "title"])
+					Local guid:String=data.GetString("guid")
+					If guid
+						Local person:TPersonBase = personCollection.GetByGUID(guid)
+						If person
+							Local personToStore:TPersonLocalization = new TPersonLocalization
+							personToStore.id = person.id
+							personToStore.firstName=data.GetString("first_name","")
+							personToStore.lastName=data.GetString("last_name","")
+							personToStore.nickName=data.GetString("nick_name","")
+							personToStore.title=data.GetString("title","")
+							if toStore.length <= index Then toStore = toStore[.. index + 50]
+							toStore[index] = personToStore
+							index:+1
+						EndIf
+					EndIf
+					dbl.persons.insert(code, toStore[..index])
```

Comment:
Kurze Verstaendnisfrage:
- es iteriert ueber "persons"-Eintraege
- fuer jeden Eintrag versucht es Eigenschaften einzulesen
- falls die in den Eigenschaften definierte Person vorhanden ist, wird "toStore" befuellt
- egal ob diese Person da ist oder nicht, die "languagecode"-Map wird mit einer aktuellen Array-Kopie befuellt

Warum machst du das nach jeder Person in der DB (unabhaengig ob je eine Person "lokalisiert" wurde?
Waere es nicht besser, `dbl.persons.insert(code, toStore[..index])` außerhalb der "For" loop zu machen?
Und dann natuerlich nur, wenn "index > 0".

Gleiches gilt fuer die Roles...

Reply:
Ja. Das finale Setzen sollte außerhalb der Schleife erfolgen.